### DISCUSSION TITLE
Hide any gallery Third-Party references (Cross-Posts) on your model page.

### DISCUSSION THREAD

**Discussion by TroubleDarkness** [civitai/civitai#419] [Ideas]
With the last update on CivitAI a new feature appeared - Cross-Post. Every post containing meta with your model name would be shown on your model page in Gallery. 

![image](https://user-images.githubusercontent.com/33435797/229874806-6235607b-d854-4173-b95c-1504b6345db9.png)

That's a cool feature, i believe. But. Some of this previews contains something like Vore, Defecation stuff or even worse. I'm kinda tolerant to this, because we live in internet and seen some shit, but someone, who just turned off nsfw filters for nudity can get not the most pleasant feelings, you know.

The main problem: There is different filters like gore, nudity and etc, but nothing about even more explicit stuff, so you can't add it to the filter. Also, i don't want someone being able to see such content on MY mix page, where I HAVE NO any authority to moderate it.  

For example, This images from Huge Two-Handed Burger LoRA

![image](https://user-images.githubusercontent.com/33435797/229875651-cd67c7fd-798c-4cc8-b5ae-f6d8b9951a65.png)

but also shown on my mix's page because of

![image](https://user-images.githubusercontent.com/33435797/229875778-769024fe-a420-49ba-aee6-c8816d90fa25.png)

So, the idea is simple, add an ability for model/mix author to hide unwanted results from model page for all users (results not related to model/mix's page). Maybe with moderator approval, maybe not. With this feature author would not be able to hide any comment/review of model, so he can't affect it's rating. Only Cross-Posts.

I never would have thought that something like this would bother me, but i've got what i've got.

**Comment by jexom** [comment]
Aye, not to kink shame and all, but some stuff catering for niche and extreme fetishes should be made possible to be filtered by the author.

**Reply by TroubleDarkness** [reply]
> Aye, not to kink shame and all, but some stuff catering for niche and extreme fetishes should be possible to be filtered by the author.

Yeah, my point is the same. I don't want so shock someone, who opens my mix page with only nudity filter off.

**Comment by TroubleDarkness** [comment]
So, no any feedback? No any comments about? 

Like "Yes" | "No" | "You idea is good, but..." | "Your idea is great, we will contact you later" and go on.

Just dead and clear nothing? 

And maybe add contol over it for very curious users like:

![image](https://user-images.githubusercontent.com/33435797/231299371-3ea5ee86-62ae-4e1b-aa8c-c3d6b5583355.png)

**Comment by JustMaier** [comment]
Hey, sorry, there's a lot of feedback here on GitHub and we haven't had time to get in here to start processing it yet and I like to be thoughtful when processing the feedback.

---

**Here's what I'm thinking:**
1. Give creators the ability to "Disavow" posts visible in their model Gallery. This will hide the post for them an anyone else viewing the page.
2. Add a "View all images using this resource" link in the gallery that links to the Image gallery filtered to images created using the resource (including disavowed content).
3. Add a filter to the model gallery to show/hide Cross-posts

**Reply by TroubleDarkness** [reply]
Yeah, that's pretty good i guess. Just posted another comment not to drown this topic, because it's already a week old. Thanks for an answer.

**Reply by JustMaier** [reply]
Anything you'd want to be different or any feedback?

**Reply by TroubleDarkness** [reply]
The "Disavow" ability, as i said, should be spread on the Cross-Posts only. Just to mention. Because hiding actual reviews is kinda... Selfish? I guess. "Oh, i don't like this review on my mix, i would hide it! C:" This is the dark side of power, we should avoid that.

Separating gallery with two buttons like "Only Reviews" and "Related Posts"/"Cross-Posts" also would be a great idea, just to give people an opportunity to seach through an actual posts without any side resources, that changes output significantly. 
or
Just make a "Show Hidden Gallery Results" button in settings like i've showed before.

This is two variants of implemeting the same feature. Not sure what would be more convinient, but any movement in this way would make CivitAI more user and creator-friendly.
